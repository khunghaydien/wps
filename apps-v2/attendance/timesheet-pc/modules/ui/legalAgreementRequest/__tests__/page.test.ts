import snapshotDiff from 'snapshot-diff';

import reducer, {
  // @ts-ignore
  __get__,
  actions,
} from '../page';

const ACTION_TYPE = __get__('ACTION_TYPE');
const initialState = __get__('initialState');

describe('reducer()', () => {
  test('@@init', () => {
    // @ts-ignore
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff({}, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.START_LOADING, () => {
    // Arrange
    const prev = initialState;
    // Act
    const next = reducer(initialState, actions.startLoading());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.FINISH_LOADING, () => {
    // Arrange
    const prev = {
      ...initialState,
      loading: true,
    };

    // Act
    const next = reducer(initialState, actions.finishLoading());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.OPEN, () => {
    // Arrange
    const prev = initialState;
    // Act
    const next = reducer(initialState, actions.openRequest());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
  test(ACTION_TYPE.CLOSE, () => {
    // Arrange
    const prev = {
      ...initialState,
      isOpen: true,
    };
    // Act
    const next = reducer(initialState, actions.closeRequest());
    // Assert
    expect(snapshotDiff(prev, next)).toMatchSnapshot();
  });
});
