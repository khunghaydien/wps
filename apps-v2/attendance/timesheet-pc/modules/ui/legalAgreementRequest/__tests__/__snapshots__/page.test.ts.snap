// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@init 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"loading\\": false,
+   \\"opened\\": false,
+ }"
`;

exports[`reducer() TIMESHEET-PC/UI/LEGAL-AGREEMENT-REQUEST/PAGE/CLOSE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"isOpen\\": true,
    \\"loading\\": false,
    \\"opened\\": false,
  }"
`;

exports[`reducer() TIMESHEET-PC/UI/LEGAL-AGREEMENT-REQUEST/PAGE/FINISH_LOADING 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"loading\\": true,
+   \\"loading\\": false,
    \\"opened\\": false,
  }"
`;

exports[`reducer() TIMESHEET-PC/UI/LEGAL-AGREEMENT-REQUEST/PAGE/OPEN 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
    \\"loading\\": false,
-   \\"opened\\": false,
+   \\"opened\\": true,
  }"
`;

exports[`reducer() TIMESHEET-PC/UI/LEGAL-AGREEMENT-REQUEST/PAGE/START_LOADING 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"loading\\": false,
+   \\"loading\\": true,
    \\"opened\\": false,
  }"
`;
